src/ts/entitysystem/components/drawing/ImageDrawable.ts

```diff
@@ -0,0 +1,168 @@
+///<reference path="./Drawable.ts"/>
+///<reference path="../../../util/JsonLoader.ts"/>
+module entityframework.components.drawing {
+
+    import observe = framework.observe;
+    import serialize = util.serialize;
+
+    /**
+     * Represents a drawable that can draw an image.
+     */
+    @serialize.ProvideClass(ImageDrawable, "ild::ImageDrawable")
+    export class ImageDrawable extends Drawable {
+        @observe.Primitive(Boolean)
+        @serialize.Ignore
+        private _loaded : boolean = false;
+
+        @observe.Object()
+        private textureRect : math.Rectangle = new math.Rectangle(0, 0, -1, -1);
+
+        @observe.Primitive(String)
+        textureKey : string = "";
+
+        @observe.Primitive(Boolean)
+        isWholeImage : boolean = true;
+
+        constructor(key : string = "") {
+            super(key);
+        }
+
+        protected generateCanvasDisplayObject(resourceManager : util.resource.ResourceManager) : createjs.DisplayObject {
+            var bitmap = null;
+            if (this.loaded) {
```

Comment:
Instead of caching _loaded on the model object it should just ask the ResourceManager if the resource has been loaded.